src/main/java/com/adyen/builders/terminal/TerminalAPIRequestBuilder.java

```diff
@@ -0,0 +1,425 @@
+import com.adyen.model.nexo.MessageClassType;
+import com.adyen.model.nexo.MessageHeader;
+import com.adyen.model.nexo.MessageType;
+import com.adyen.model.nexo.PINRequest;
+import com.adyen.model.nexo.PaymentRequest;
+import com.adyen.model.nexo.PrintRequest;
+import com.adyen.model.nexo.ReconciliationRequest;
+import com.adyen.model.nexo.ReversalRequest;
+import com.adyen.model.nexo.SaleToPOIRequest;
+import com.adyen.model.nexo.SoundRequest;
+import com.adyen.model.nexo.StoredValueRequest;
+import com.adyen.model.nexo.TransactionStatusRequest;
+import com.adyen.model.nexo.TransmitRequest;
+import com.adyen.model.terminal.TerminalAPIRequest;
+
+import static com.adyen.constants.ApiConstants.TerminalAPI.PROTOCOL_VERSION;
+
+public final class TerminalAPIRequestBuilder {
+
+    // MessageHeader
+    private String saleId;
+    private String serviceId;
+    private String poiId;
+    private MessageClassType messageClass;
+    private MessageCategoryType messageCategory;
+
+    // SaleToPOIRequest
+    private AbortRequest abortRequest;
+    private BalanceInquiryRequest balanceInquiryRequest;
+    private BatchRequest batchRequest;
+    private CardAcquisitionRequest cardAcquisitionRequest;
+    private AdminRequest adminRequest;
+    private DiagnosisRequest diagnosisRequest;
+    private DisplayRequest displayRequest;
+    private EnableServiceRequest enableServiceRequest;
+    private EventNotification eventNotification;
+    private GetTotalsRequest getTotalsRequest;
+    private InputRequest inputRequest;
+    private InputUpdate inputUpdate;
+    private LoginRequest loginRequest;
+    private LogoutRequest logoutRequest;
+    private LoyaltyRequest loyaltyRequest;
+    private PaymentRequest paymentRequest;
+    private PINRequest pinRequest;
+    private PrintRequest printRequest;
+    private CardReaderInitRequest cardReaderInitRequest;
+    private CardReaderAPDURequest cardReaderAPDURequest;
+    private CardReaderPowerOffRequest cardReaderPowerOffRequest;
+    private ReconciliationRequest reconciliationRequest;
+    private ReversalRequest reversalRequest;
+    private SoundRequest soundRequest;
+    private StoredValueRequest storedValueRequest;
+    private TransactionStatusRequest transactionStatusRequest;
+    private TransmitRequest transmitRequest;
+    private ContentInformation securityTrailer;
+
+    public TerminalAPIRequestBuilder() {
+    }
+
+    public TerminalAPIRequestBuilder(String saleId, String serviceId, String poiId) {
+        this.saleId = saleId;
+        this.serviceId = serviceId;
+        this.poiId = poiId;
+    }
+
+    public TerminalAPIRequestBuilder withSaleId(String saleId) {
+        this.saleId = saleId;
+        return this;
+    }
+
+    public TerminalAPIRequestBuilder withServiceId(String serviceId) {
+        this.serviceId = serviceId;
+        return this;
+    }
+
+    public TerminalAPIRequestBuilder withPoiId(String poiId) {
+        this.poiId = poiId;
+        return this;
+    }
```

Comment:
remove this.

Reply:
Removed, kept it only on the constructor